File: examples/Ruby/interrupt.rb
# Shows how to handle Ctrl-C
require 'ffi-rzmq'

context = ZMQ::Context.new(1)
socket = context.socket(ZMQ::REP)
socket.bind("tcp://*:5558")

trap("INT") { puts "Shutting down."; socket.close; context.terminate; exit}

puts "Starting up"

while true do
  message = socket.recv_string
  puts "Message: #{message.inspect}"
  socket.send_string("Message received")
end

File: examples/Node.js/interrupt.js
// Show how to handle Ctrl+C in Node.js

var zmq = require('zmq')
  , socket = zmq.createSocket('rep');

socket.on('message', function(buf) {
  // echo request back
  socket.send(buf);
});

process.on('SIGINT', function() {
  socket.close();
  process.exit();
});

socket.bindSync('tcp://*:5555');
